gateway-service-knoxtoken/src/main/java/org/apache/knox/gateway/service/knoxtoken/TokenResource.java

```diff
@@ -246,9 +249,14 @@ public Response renew(String token) {
       String renewer = SubjectUtils.getCurrentEffectivePrincipalName();
       if (allowedRenewers.contains(renewer)) {
         try {
+          JWTToken jwt = new JWTToken(token);
           // If renewal fails, it should be an exception
-          expiration = tokenStateService.renewToken(token,
+          expiration = tokenStateService.renewToken(jwt,
                                                     renewInterval.orElse(tokenStateService.getDefaultRenewInterval()));
+          log.renewedToken(getTopologyName(), TokenUtils.getTokenDisplayText(token), TokenUtils.getTokenId(jwt));
+        } catch (ParseException e) {
+          log.invalidToken(getTopologyName(), TokenUtils.getTokenDisplayText(token), e);
+          error = e.getMessage();
```

Comment:
e.getMessage() will throw NPE in case message is null.

Reply:
Yes, getMessage() may return null. I will add logic to handle that, but I don't believe replacing it with toString() is equivalent.